packages/core/src/components/overlay2/overlay2.tsx

```diff
@@ -325,14 +306,41 @@ export const Overlay2 = React.forwardRef<OverlayInstance, Overlay2Props>((props,
         }
     }, [isOpen, overlayWillOpen, overlayWillClose, prevIsOpen]);
 
-    // run once on unmount
+    // Important: clean up old document-level event listeners if their memoized values change (this is rare, but
```

Comment:
I'm not sure if this order matters - I moved these below the `useEffect` that calls `overlayWillOpen` because these event listeners were attached towards the end of that method and I believe `useEffect` hooks are called in order. But now I'm seeing these were also attached before `setRef(lastActiveElementBeforeOpened, getActiveElement(getRef(containerElement)));` and now will be after.

I don't expect that to be an issue, but also don't see any issues if these get attached before the other code in `overlayWillOpen` runs.